### ISSUE TITLE
Improve rich notifications by guessing shutter position on state transitions

### ISSUE THREAD

**Issue by nzapponi** [nzapponi/homebridge-simplisafe3#47]
During alarm state transitions, the shutter is open. We should allow the plugin to pull snapshots when that is the case.

**Comment by nikonratm** [comment]
I see you’re a bit of a perfectionist =) I can take care of this unless you’re already working on it?

But I have to say, I intentionally coded the [switch statement](https://github.com/nzapponi/homebridge-simplisafe3/blob/a882e2991bf6ab98b41ae74073348ddf5ca8f9bc/src/accessories/simplicam.js#L210) with no default case in order to cover most these situations. E.g. when you come home and the alarm goes off for a few seconds it still works (probably the most important one) or when arming for away. Correct me if you’re seeing something different but the only situation this doesn’t currently work is when you’re disarming the alarm without triggering it. Since SimpliSafe state is “OFF” in that case it would just have to be a 10 second timer. This kinda seemed like the least important situation to me but I guess I can imagine someone wanting the visual notification when someone disables their alarm and it is a ‘behavior’ of the camera so might as well support it.

Anyway, let me know if you’ve done it already otherwise I can take care of it soon, I do feel a little sense of ownership over this feature 🙂

**Comment by nzapponi** [comment]
Yeah that’s right, it’s not a high priority feature since it only applies when disarming the alarm without triggering it. I find myself with notifications from the alarm saying “the camera detected motion” but then no way of seeing the video feed. 

Again, an edge case, but still something we might want to add in for the sake of it. 

I wouldn’t use a timer though, as the shutter stays open for “some time” after disarming, until no more motion is detected.

I was thinking of using a logic like “when a CAMERA_MOTION event is received, load a snapshot regardless of the shutter privacy options for 5 seconds or so”, since obviously if we are receiving that event from the camera, the privacy shutter *must* be open. 

It’d be great if you wanted to take a look into this! Otherwise I can do it over the weekend, I don’t mind. 

Thanks!

**Comment by nikonratm** [comment]
I created a PR for this that does implement the functionality logically, just one note below. The idea is to have snapshot request just bypass the privacy shutter logic when the request is the result of a motion sensor trigger, like you mentioned. The way I achieved it was with a dead-simple additional property `motionIsTriggered` on cameraSource. This property is set when the motion event is received and reset according to the 5 second timer we use to reset the motion sensor but then obviously gets set to true again next time motion is detected. Totally works.

But perhaps the best way would be to just check the motion sensor's `Characteristic.MotionDetected` since that is essentially the same as the `cameraSource .motionIsTriggered` property. But the cameraSource object doesnt maintain a reference to the camera accessory so rather than bloat cameraSource with a reference we'll only ever use for one tiny thing I figured just keep it simple with the boolean solution. Let me know if you think we should go the other way...
